packages/dynamodb-emulator/client.js

```diff
@@ -0,0 +1,145 @@
+// client lock aquisition dance ....
+const objHash = require('object-hash');
+const path = require('path');
+const http = require('http');
+const fs = require('fs');
+const qs = require('querystring');
+const { flockSync } = require('fs-ext');
+const { spawn } = require('child_process');
+const log = require('logdown')('dynamodb-emulator:client');
+const { getClient } = require('./index');
+const { homedir } = require('os');
+
+const homeDir = homedir();
+const waitTimeout = 30 * 1000;
+const requestTimeout = waitTimeout;
+const deamonPath = path.join(__dirname, 'deamon.js');
+const pidPath = path.join(homeDir, '.pid');
+const wait = ms =>
+  new Promise(accept => {
+    setTimeout(accept, ms);
+  });
+const fsExists = file =>
+  new Promise(accept => {
+    fs.exists(file, accept);
+  });
+const get = (unixSocket, urlpath, params = {}) =>
+  new Promise((accept, reject) => {
+    log.info('request', urlpath, params);
+    const start = Date.now();
+    const req = http.request(
+      {
+        socketPath: unixSocket,
+        timeout: requestTimeout,
+        method: 'GET',
+        path: `/${urlpath}?${qs.stringify(params)}`,
+        headers: {
+          'Content-Length': 0,
+        },
+      },
+      res => {
+        res.setEncoding('utf8');
```

Comment:
set encoding should cause everything to return a string right ?

Reply:
🤦‍♂️ 👍